index.js

```diff
@@ -56,6 +56,25 @@ module.exports = {
     'import/no-named-default': 'error',
     'import/no-webpack-loader-syntax': 'error',
     'import/prefer-default-export': 'error',
+    'jsdoc/check-examples': 0,
+    'jsdoc/check-param-names': 1,
+    'jsdoc/check-tag-names': 1,
+    'jsdoc/check-types': 1,
+    'jsdoc/newline-after-description': 0,
+    'jsdoc/no-undefined-types': 0, // did not consider InputValidationError as valid type
+    'jsdoc/require-description': 0,
+    'jsdoc/require-description-complete-sentence': 0,
+    'jsdoc/require-example': 0,
+    'jsdoc/require-hyphen-before-param-description': 0,
+    'jsdoc/require-param': 1,
+    'jsdoc/require-param-description': 0,
+    'jsdoc/require-param-name': 1,
+    'jsdoc/require-param-type': 1,
+    'jsdoc/require-returns': 1,
```

Comment:
Hopefull it won't require `@returns` if we use some early returns (e.g. `return;`) in functions which are not returning anything

Reply:
This requires that return type must be always documented even if it would be `void`/`undefined`. Typescript compiler only understands `void` but jsdoc plugin only allows `undefined`.
As a fix I will propose keeping this setting and removing `require-returns-type` so that it works with static type checking.

Reply:
fixed in the way I proposed

Reply:
From what I see it'll be fixed with near version of plugin -> https://github.com/gajus/eslint-plugin-jsdoc/issues/142 (I guess will apply with next version_

So I guess we may keep `jsdoc/require-returns` together with `jsdoc/require-type`

Reply:
excellent! I think we can wait a bit for this, this is among the top 2 issues in the codebase.